### PR TITLE
ZIPPYDOC (Certified-Connector)

### PR DESCRIPTION
**Pull request by worm333**
---
### When submitting a connector, please make sure that you follow the requirements below, otherwise your PR might be rejected. We want to make you have a well-built connector, a smooth certification experience, and your users are happy :) 

If this is your first time submitting to GitHub and you need some help, please sign up for this [session](https://forms.office.com/pages/responsepage.aspx?id=KtIy2vgLW0SOgZbwvQuRaXDXyCl9DkBHq4A2OG7uLpdUMTFJWFFGVUxBNUFZQjZWRUdaOE5BMFkwNS4u). 

- [x] I attest that the connector doesn't exist on the Power Platform today. I've verified by checking the pull requests in GitHub and by searching for the connector on the platform or in the documentation. 
- [x] I attest that the connector works and I verified by deploying and testing all the operations. 
- [x] I attest that I have added detailed descriptions for all operations and parameters in the swagger file.
- [x] I attest that I have added response schemas to my actions, unless the response schema is dynamic. 
- [x] I validated the swagger file, `apiDefinition.swagger.json`, by running `paconn validate` command.
- [x] If this is a certified connector, I confirm that `apiProperties.json` has a valid brand color and doesn't use an invalid brand color, `#007ee5` or `#ffffff`. If this is an independent publisher connector, I confirm that I am not submitting a connector icon.

If you are an Independent Publisher, you must also attest to the following to ensure a smooth publishing process:
- [ ] I have named this PR after the pattern of "Connector Name (Independent Publisher)" ex: HubSpot Marketing (Independent Publisher)
- [ ] Within this PR markdown file, I have pasted screenshots that show: 3 unique operations (actions/triggers) working within a Flow. This can be in one flow or part of multiple flows. For each one of those flows, I have pasted in screenshots of the Flow succeeding. 
- [ ] Within this PR markdown file, I have pasted in a screenshot from the Test operations section within the Custom Connector UI.  
- [ ] If the connector uses OAuth, I have provided detailed steps on how to create an app in the readme.md.

### CODE REVIEW

**Comment by v-EgorKozhadei** [comment]
Hi @worm333!

Thank you very much for your submission!

Could you please tell me if you are a verified or an independent publisher?
Your PR name says that it's an independent publisher, but in your apiDefinition.swagger.json file the contact information section contains information about a company. We need to make sure that everything is fine before starting the review.

Let me know if you have any questions!

**Comment by Amjed-Ayoub** [comment]
Hello @worm333,
I hope you are doing well. 
Can you please confirm if you own the API for your connector?
I look forward to hearing from you.
Please let me know if you have any questions
Thank you very much.

**Comment by worm333** [comment]
Hi @Amjed-Ayoub and @v-EgorKozhadei, I've mistakenly assumed that we need to publish connector as independent publisher, however in fact we want to publish certified connector. I've already updated pull request, please let me know if any changes still need to be made. 

Thanks

**Comment by Amjed-Ayoub** [comment]
Thank you so much @worm333 for the confirmation. Our Engineers will review your pull request very soon.
Please let us know if you have any questions.
Thank you again for your hard work.

File: certified-connectors/Zippydoc/README.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+## ZIPPYDOC Connector
+ZIPPYDOC allows to merge, transform, filter, extract, report and analyze structured and table based data and to flexibly create data flows. Our connector helps to integrate ZIPPYDOC with third party services.
+
+
```
Comment (v-EgorKozhadei) [inline certified-connectors/Zippydoc/README.md:4]: @worm333 please format the readme.md file according to [the template](https://github.com/microsoft/PowerPlatformConnectors/blob/dev/templates/certified-connectors/readme.md?plain=1). All of the required headings should be present.

File: certified-connectors/Zippydoc/apiDefinition.swagger.json [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+    },
+    "/document/power-automate/{name}": {
+      "post": {
+        "consumes": [
+          "application/json"
+        ],
+        "responses": {
+          "default": {
+            "description": "default",
+            "schema": {}
```
Comment (v-EgorKozhadei) [inline certified-connectors/Zippydoc/apiDefinition.swagger.json:113]: @worm333 using default responses with empty schemas is [not allowed](https://docs.microsoft.com/en-us/connectors/custom-connectors/certification-submission#define-exact-operation-responses). You could use 2xx response with "OK" description if it's a successful operation. Could you please define exact response schemas only for actions in your operations? You can refer to [this example](https://docs.microsoft.com/en-us/connectors/custom-connectors/coding-standards#responses).
For triggers everything is fine.

File: certified-connectors/Zippydoc/apiDefinition.swagger.json [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+            "required": true,
+            "schema": {
+              "type": "object",
+              "properties": {
+                "x-ms-notification-url": {
+                  "type": "string",
+                  "description": "x-ms-notification-url",
+                  "x-ms-notification-url": true,
+                  "x-ms-visibility": "internal",
+                  "title": ""
```
Comment (v-EgorKozhadei) [inline certified-connectors/Zippydoc/apiDefinition.swagger.json:63]: @worm333 please fill in empty titles. I know that this operation is internal, but empty properties are not allowed.

File: certified-connectors/Zippydoc/apiDefinition.swagger.json [hunk 4]
```diff
@@ -0,0 +1,258 @@
...
+                },
+                "type": {
+                  "type": "string",
+                  "description": "type",
+                  "title": "",
```
Comment (v-EgorKozhadei) [inline certified-connectors/Zippydoc/apiDefinition.swagger.json:68]: @worm333 this one too.

File: certified-connectors/Zippydoc/apiDefinition.swagger.json [hunk 5]
```diff
@@ -0,0 +1,258 @@
...
+    },
+    "/excel/table/power-automate/upload": {
+      "post": {
+        "consumes": [
+          "application/json"
+        ],
+        "responses": {
+          "default": {
+            "description": "default",
+            "schema": {}
```
Comment (v-EgorKozhadei) [inline certified-connectors/Zippydoc/apiDefinition.swagger.json:140]: @worm333 mentioned in the comment above. Actions in your operations should not have default responses with an empty schema. Please define exact responses for your actions.

File: certified-connectors/Zippydoc/apiDefinition.swagger.json [hunk 6]
```diff
@@ -0,0 +1,258 @@
...
+          }
+        ]
+      }
+    },
+    "/excel/table/power-automate/id/{tableId}": {
+      "get": {
+        "responses": {
+          "default": {
+            "description": "default",
+            "schema": {}
```
Comment (v-EgorKozhadei) [inline certified-connectors/Zippydoc/apiDefinition.swagger.json:183]: @worm333 mentioned above.

File: certified-connectors/Zippydoc/apiDefinition.swagger.json [hunk 7]
```diff
@@ -0,0 +1,258 @@
...
+          }
+        ]
+      }
+    },
+    "/excel/table/power-automate/name/{name}": {
+      "get": {
+        "responses": {
+          "default": {
+            "description": "default",
+            "schema": {}
```
Comment (v-EgorKozhadei) [inline certified-connectors/Zippydoc/apiDefinition.swagger.json:208]: @worm333 mentioned above.

**Review by v-EgorKozhadei — COMMENTED** [review]
Hi @worm333!
Please check comments after the review.
Let me know if you have any questions, thank you!

**Comment by v-EgorKozhadei** [comment]
Hello @worm333, 

I hope you are doing well. 

I am following up with you to see how things are going with preparing your connector. 

Please let us know if you have any questions. 

Thank you.

**Comment by worm333** [comment]
Hi @v-EgorKozhadei , 

I've made following changes:
- adjusted readme.md with accordance to the template
- defined response schemas 
- filled in empty titles

Please let me know if there is anything needs to be added.

Thanks.

File: certified-connectors/Zippydoc/apiDefinition.swagger.json [hunk 8]
```diff
@@ -0,0 +1,331 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "ZIPPYDOC",
+    "description": "ZIPPYDOC is a platform that allows to merge, transform, filter, extract, report and analyze structured and table based data and to flexibly create data flows",
```
Comment (v-EgorKozhadei) [inline certified-connectors/Zippydoc/apiDefinition.swagger.json:5]: @worm333 could you please change the description just a little bit? It should be concise and reflect the main value and purpose offered by your connector.
More information about the description can be found [here](https://docs.microsoft.com/en-us/connectors/custom-connectors/certification-submission#write-a-description-for-your-connector).

**Review by v-EgorKozhadei — COMMENTED** [review]
Hi @worm333!

Thank you very much for an update!
There's one thing that we can change before proceeding to the next step.

File: certified-connectors/Zippydoc/apiDefinition.swagger.json (hunk 8 above)
Comment (worm333) [reply certified-connectors/Zippydoc/apiDefinition.swagger.json]: Hi @v-EgorKozhadei , I've updated description, so it describes in more detail the main purpose and value offered by ZIPPYDOC

**Review by v-EgorKozhadei — APPROVED** [review]

**Comment by v-EgorKozhadei** [comment]
Hello @worm333, 

I hope you are doing well. 

Congratulations, your pull request is approved and merged. Please [submit the files to ISV Studio](https://docs.microsoft.com/en-us/connectors/custom-connectors/submit-for-certification#submit-to-isv-studio) or [update your submission](https://docs.microsoft.com/en-us/connectors/custom-connectors/certification-updates) using your latest commit ID, please allow us up to 1-2 weeks to review your submission, and our Engineers will notify you in the “Activity Control” area in ISV Studio. Please [make sure that the icon meets all the requirements](https://docs.microsoft.com/en-us/connectors/custom-connectors/certification-submission#design-an-icon-for-your-connector), that it’s size is 230x230, icon background is not white or transparent and it matches the iconBrandColor property in apiProperties.json file. 

After making the submission in ISV Studio, please attach an intro.md file in the activity control section. This file is required and it's different from readme.md. You can [find an example of intro.md file on this page](https://docs.microsoft.com/en-us/connectors/custom-connectors/submit-for-certification#submit-to-isv-studio) at step 6. 

Please [Create an environment in the Preview region](https://docs.microsoft.com/en-us/connectors/custom-connectors/certification-testing#create-an-environment-in-the-preview-region). You will use this environment later to [test your connector](https://docs.microsoft.com/en-us/connectors/custom-connectors/certification-testing#test-your-connector) after Microsoft is done with the functional verification for your connector.  

We expect all tests to be completed within 1-2 weeks. 

If your connector passes all tests, the deployment process begins, and it typically takes up to 3 to 4 weeks to deploy your connector to all of our regions. 

Please let me know if you have any questions. 

Thank you very much for working with us.

**Pull request merged**